src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,888 @@
+
+		protected override void OnPointerReleased(PointerRoutedEventArgs args)
+		{
+			base.OnPointerReleased(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnPointerReleased.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnPointerReleased(args);
+
+		}
+
+
+		protected override void OnGotFocus(RoutedEventArgs args)
+		{
+			base.OnGotFocus(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnGotFocus.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnGotFocus(args);
+		}
+
+
+		protected override void OnLostFocus(RoutedEventArgs args)
+		{
+			base.OnLostFocus(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnLostFocus.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnLostFocus(args);
+		}
+
+		// KeyDown event handler that handles the keyboard navigation between
+		// the menu items and shows the MenuFlyoutSubItem in case of hitting
+		// the enter or right arrow key.
+
+		protected override void OnKeyDown(KeyRoutedEventArgs args)
+		{
+			base.OnKeyDown(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnKeyDown.", this));
+#endif // MFSI_DEBUG
+
+			bool handled = args.Handled;
+			bool shouldHandleEvent = false;
+
+			if (!handled)
+			{
+				MenuFlyoutPresenter spParentPresenter = GetParentMenuFlyoutPresenter();
+
+				if (spParentPresenter != null)
+				{
+					var key = VirtualKey.None;
+
+					key = args.Key;
+
+					// Navigate each item with the arrow down or up key
+					if (key == VirtualKey.Down || key == VirtualKey.Up)
+					{
+						spParentPresenter.HandleUpOrDownKey(key == VirtualKey.Down);
+						UpdateVisualState();
+
+						// If we handle the event here, it won't get handled in m_menuHelper.OnKeyDown,
+						// so we'll do that afterwards.
+						shouldHandleEvent = true;
+					}
+				}
+			}
+
+			m_menuHelper.OnKeyDown(args);
+			args.Handled = shouldHandleEvent;
+		}
+
+
+		protected override void OnKeyUp(KeyRoutedEventArgs args)
+		{
+			base.OnKeyUp(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnKeyUp.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnKeyUp(args);
+		}
+
+		// Ensure the creating the popup and menu presenter to show the 
+		void EnsurePopupAndPresenter()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: EnsurePopupAndPresenter.", this));
+#endif // MFSI_DEBUG
+
+			if (m_tpPopup == null)
+			{
+				MenuFlyoutPresenter spParentMenuFlyoutPresenter = null;
+				Control spPresenter;
+				UIElement spPresenterAsUI;
+				FrameworkElement spPresenterAsFE;
+				Popup spPopup;
+
+				spPopup = new Popup();
+				spPopup.IsSubMenu = true;
+				spPopup.IsLightDismissEnabled = false;
+
+				spParentMenuFlyoutPresenter = GetParentMenuFlyoutPresenter();
+				if (spParentMenuFlyoutPresenter != null)
+				{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either remove or fill this block of code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)